File: 11389 The Bus Driver Problem/solution.java
import java.io.*;
import java.util.*;

public class Main {
	public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StreamTokenizer st = new StreamTokenizer(br);
        StringBuilder output = new StringBuilder();
        while(st.nextToken() == StreamTokenizer.TT_NUMBER) {
            int n = (int)st.nval;
            if(n == 0)
                break;
            st.nextToken();
            int d = (int)st.nval;
            st.nextToken();
            int r = (int)st.nval;
            short[] morning = new short[n], evening = new short[n];
            for(int i = 0; i < n; ++i) {
                st.nextToken();
                morning[i] = (short)st.nval;
            }
            for(int i = 0; i < n; ++i) {
                st.nextToken();
                evening[i] = (short)st.nval;
            }
            Arrays.sort(morning);
            Arrays.sort(evening);
            int pay = 0;
            for(int i = 0; i < n; ++i) {
                short route = (short)(morning[i] + evening[n - 1 - i]);
                pay += (route > d) ? (route - d) * r : 0;
            }
            output.append(pay).append('\n');
        }
        System.out.print(output);
	}
}

File: 11389 The Bus Driver Problem/solution.c
#include <stdio.h>
#include <stdlib.h>

int compare(const void *a, const void *b) {
    return *(short*)a > *(short*)b;
}

int main() {
    int n, d, r;
    while(scanf("%d %d %d", &n, &d, &r) && n != 0) {
        short morning[n], evening[n];
        for(int i = 0; i < n; ++i)
            scanf("%hd", &morning[i]);
        for(int i = 0; i < n; ++i)
            scanf("%hd", &evening[i]);
        qsort(morning, n, sizeof(short), compare);
        qsort(evening, n, sizeof(short), compare);
        int pay = 0;
        for(int i = 0; i < n; ++i) {
            short route = morning[i] + evening[n - 1 - i];
            pay += (route > d) ? (route - d) * r : 0;
        }
        printf("%d\n", pay);
    }
    return 0;
}
